packages/flutter/lib/src/services/text_input.dart

```diff
@@ -852,25 +856,43 @@ abstract class TextInputClient {
   ///
   /// [TextInputClient] should cleanup its connection and finalize editing.
   void connectionClosed();
+
+  /// Framework notified of a restored connection.
+  ///
+  /// [TextInputClient] should show the text input control if appropriate.
+  void connectionRestored();
```

Comment:
Unsure about the naming here. The client needs to be notified in a way or another when a connection is replaced on the fly.